rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpace.scala

```diff
@@ -90,18 +85,11 @@ object ChargingRSpace {
         result match {
           case Left(oope) =>
             // if we run out of phlos during the match we have to zero phlos available
-            costAlg.get().flatMap(cost => costAlg.charge(cost)) >> Sync[F].raiseError(oope)
+            cost.get.flatMap(charge[F]) >> Sync[F].raiseError(oope)
```

Comment:
there's so much self-contradictory in this line... the cost should be negative already if it's oope, we shouldn't be charging for "all there's left", because there's none already, and we could probably (hopefully?) replace this with just `rethrow(result)`. See my other comment though: https://github.com/rchain/rchain/pull/2251#discussion_r264175656

Reply:
Yes. It's redundant now. I'll refactor to just re-throw the error. That's the best that can be done until `Either` is eliminated from `ISpace`.